test/karma/directives/routeOnClick.test.js

```diff
@@ -22,13 +22,51 @@ describe('routeOnClick directive', function() {
       });
     });
   });
+
+  describe('when element is an Anchor tag', function() {
+    it('should populate the href attribute of the element with the URL returned from the matching URL writer', function () {
+      setupMockUrlWriter();
+      inject(function ($rootScope, $compile) {
+        let element = $compile('<a route-on-click="routeLinkUrlWriter()">Link</a>')($rootScope.$new());
+        $rootScope.$digest();
+        expect(element.attr('href')).toBe('#/contacts/update/1');
+      });
+    });
+
+    it('should update the href attribute when the expression eval changes', function () {
+      window.angular.mock.module(function(RouteProvider) {
+        RouteProvider.setPersistentStates('page');
+        RouteProvider.setHtml5Mode(false);
+        RouteProvider.registerUrlWriter('pagination', function() {
+          return '/page/2';
+        });
+      });
+
+      inject(function ($rootScope, $compile, State) {
+        State.set('page', 2);
+        let element = $compile('<a route-on-click="paginationUrlWriter()">Link</a>')($rootScope.$new());
+        $rootScope.$digest();
+        expect(element.attr('href')).toBe('#/page/2');
+
+        $rootScope.paginationUrlWriter = function() {
+          return '/page/3';
+        };
```

Comment:
I'm now not a big fan of the $rootScope stuff, do you think we should upgrade this